bugsnag-plugin-android-anr/src/main/java/com/bugsnag/android/AnrDetailsCollector.kt

```diff
@@ -0,0 +1,73 @@
+package com.bugsnag.android
+
+import android.app.ActivityManager
+import android.app.ActivityManager.ProcessErrorStateInfo
+import android.content.Context
+import android.os.Handler
+import android.os.HandlerThread
+import android.os.Process
+import android.support.annotation.VisibleForTesting
+import java.util.concurrent.atomic.AtomicInteger
+
+internal class AnrDetailsCollector {
+
+    companion object {
+        private const val INFO_POLL_THRESHOLD_MS: Long = 1000
+        private const val MAX_ATTEMPTS: Int = 30
+    }
+
+    private val handlerThread = HandlerThread("bugsnag-anr-collector")
+
+    init {
+        handlerThread.start()
+    }
+
+    fun collectAnrDetails(ctx: Context): ProcessErrorStateInfo? {
+        val am = ctx.getSystemService(Context.ACTIVITY_SERVICE) as ActivityManager
+        return captureProcessErrorState(am, Process.myPid())
+    }
+
+    /**
+     * Collects information about an ANR, by querying an activity manager for information about
+     * any proceses which are currently in an error condition.
+     *
+     * See https://developer.android.com/reference/android/app/ActivityManager.html#getProcessesInErrorState()
+     */
+    @VisibleForTesting
+    internal fun captureProcessErrorState(am: ActivityManager, pid: Int): ProcessErrorStateInfo? {
+        return try {
+            val processes = am.processesInErrorState ?: emptyList()
+            processes.firstOrNull { it.pid == pid }
+        } catch (exc: RuntimeException) {
+            null
+        }
+    }
+
+    internal fun mutateError(error: Error, anrState: ProcessErrorStateInfo) {
```

Comment:
This function could use a more descriptive name about how the error is changed. `addErrorStateInfo()`, etc.